test/e2e/query_test.go

```diff
@@ -166,23 +167,85 @@ func TestQuery(t *testing.T) {
 	})
 }
 
-func TestQueryRoutePrefix(t *testing.T) {
+func TestQueryExternalPrefixWithoutReverseProxy(t *testing.T) {
 	t.Parallel()
 
 	s, err := e2e.NewScenario("e2e_test_query_route_prefix")
 	testutil.Ok(t, err)
 	t.Cleanup(e2ethanos.CleanScenario(t, s))
 
+	externalPrefix := "test"
+
 	q, err := e2ethanos.NewQuerier(
 		s.SharedDir(), "1",
 		nil,
 		nil,
 		nil,
-		"test",
+		"",
+		externalPrefix,
 	)
 	testutil.Ok(t, err)
 	testutil.Ok(t, s.StartAndWaitReady(q))
 
+	checkNetworkRequests(t, "http://"+q.HTTPEndpoint()+"/"+externalPrefix+"/graph")
+}
+
+func TestQueryExternalPrefix(t *testing.T) {
+	t.Parallel()
+
+	s, err := e2e.NewScenario("e2e_test_query_external_prefix")
+	testutil.Ok(t, err)
+	t.Cleanup(e2ethanos.CleanScenario(t, s))
+
+	externalPrefix := "thanos"
+
+	q, err := e2ethanos.NewQuerier(
+		s.SharedDir(), "1",
+		nil,
+		nil,
+		nil,
+		"",
+		externalPrefix,
+	)
+	testutil.Ok(t, err)
+	testutil.Ok(t, s.StartAndWaitReady(q))
+
+	querierURL := urlParse(t, "http://"+q.HTTPEndpoint()+"/"+externalPrefix)
+
+	querierProxy := httptest.NewServer(e2ethanos.NewSingleHostReverseProxy(querierURL, externalPrefix))
```

Comment:
Let's add `defer querierProxy.Close()` here

Reply:
Gotcha. Also a question, can we use `t.Cleanup` here instead of `defer`?

Reply:
Sure, that's even better.

Reply:
Fixed this :)